src/test/scala/application/actors/CustomerServerActorTest.scala

```diff
@@ -0,0 +1,260 @@
+    describe("when first started up") {
+      it("should notify it to the root actor") {
+        rootActorProbe.expectMessage(10.seconds, Startup(true))
+      }
+    }
+  }
+
+  describe("A customer") {
+    describe("after being registered") {
+      it("should be present into the database") {
+        val server: ActorRef[CustomerServerCommand] = customerServer.getOrElse(fail())
+        server ! RegisterCustomer(customer, password, customerResponseProbe.ref)
+        customerResponseProbe.expectMessage(10.seconds, CustomerResponse(Right[ValidationError, Customer](customer)))
+        server ! LoginCustomer(email, password, customerResponseProbe.ref)
+        customerResponseProbe.expectMessage(10.seconds, CustomerResponse(Right[ValidationError, Customer](customer)))
+        server ! DeregisterCustomer(email, password, emptyResponseProbe.ref)
+        emptyResponseProbe.expectMessage(10.seconds, EmptyResponse(Right[ValidationError, Unit](())))
+        messageBrokerProbe.expectMessage(10.seconds, CustomerUnregistered(email))
+      }
+    }
+
+    describe("after being registered and while trying to login with the wrong password") {
+      it("should not be allowed") {
+        val server: ActorRef[CustomerServerCommand] = customerServer.getOrElse(fail())
+        server ! RegisterCustomer(customer, password, customerResponseProbe.ref)
+        customerResponseProbe.expectMessage(10.seconds, CustomerResponse(Right[ValidationError, Customer](customer)))
+        server ! LoginCustomer(email, otherPassword, customerResponseProbe.ref)
+        customerResponseProbe.expectMessage(10.seconds, CustomerResponse(Left[ValidationError, Customer](PasswordNotMatching)))
+        server ! DeregisterCustomer(email, password, emptyResponseProbe.ref)
+        emptyResponseProbe.expectMessage(10.seconds, EmptyResponse(Right[ValidationError, Unit](())))
+        messageBrokerProbe.expectMessage(10.seconds, CustomerUnregistered(email))
+      }
+    }
+
+    describe("after being registered and then deleted") {
+      it("should not be present into the database") {
+        val server: ActorRef[CustomerServerCommand] = customerServer.getOrElse(fail())
+        server ! RegisterCustomer(customer, password, customerResponseProbe.ref)
+        customerResponseProbe.expectMessage(10.seconds, CustomerResponse(Right[ValidationError, Customer](customer)))
+        server ! DeregisterCustomer(email, password, emptyResponseProbe.ref)
+        emptyResponseProbe.expectMessage(10.seconds, EmptyResponse(Right[ValidationError, Unit](())))
+        messageBrokerProbe.expectMessage(10.seconds, CustomerUnregistered(email))
+        server ! LoginCustomer(email, password, customerResponseProbe.ref)
+        customerResponseProbe.expectMessage(10.seconds, CustomerResponse(Left[ValidationError, Customer](CustomerNotFound)))
+      }
+    }
+
+    describe("after being registered and while trying to being deleted with the wrong password") {
```

Comment:
"after being registered and while trying to ( _get deleted_ or _deregister_ ) with the wrong password"